src/main/java/org/kumoricon/registration/print/formatter/badgeimage/BadgeImage.java

```diff
@@ -39,27 +40,66 @@ public class BadgeImage {
     }
 
-    void drawStretchedImage(Image image, Rectangle area) {
+    void drawStretchedImage(Image image, Rectangle boundingArea) {
         if (image == null) return;
-        double imageWidth = image.getWidth(null);
-        double imageHeight = image.getHeight(null);
-
-        double widthRatio = area.getWidth() / imageWidth;
-        double heightRatio = area.getHeight() / imageHeight;
-        double ratio = Math.min(widthRatio, heightRatio);
-
-        int newWidth = (int) (imageWidth * ratio);
-        int newHeight = (int) (imageHeight * ratio);
-
-        Rectangle scaledArea = new Rectangle(
-                area.x + (area.width - newWidth)/2,
-                area.y + (area.height - newHeight)/2,
-                newWidth,
-                newHeight);
-        drawImage(image, scaledArea);
+        int imageWidth = image.getWidth(null);
+        int imageHeight = image.getHeight(null);
+
+        Dimension imageDimension = new Dimension(imageWidth, imageHeight);
+        Dimension boundaryDimension = new Dimension(boundingArea.width, boundingArea.height);
+        Dimension scaledDimension = getScaledDimension(imageDimension, boundaryDimension);
+
+        Rectangle imageArea = new Rectangle(
+            boundingArea.x,
+            boundingArea.y,
+            scaledDimension.width,
+            scaledDimension.height
+        );
+
+        int heightDifference = boundingArea.height - imageArea.height;
+        int widthDifference = boundingArea.width - imageArea.width;
+
+        Rectangle centeredImage = new Rectangle(
+            boundingArea.x + (widthDifference/2),
+            boundingArea.y + (heightDifference/2),
+            imageArea.width,
+            imageArea.height
+        );
+
+        drawImage(image, centeredImage);
+    }
+
+    public static Dimension getScaledDimension(Dimension imgSize, Dimension boundary) {
+        int original_width = imgSize.width;
+        int original_height = imgSize.height;
+        int bound_width = boundary.width;
+        int bound_height = boundary.height;
+        int new_width = original_width;
+        int new_height = original_height;
+    
+        // first check if we need to scale width
+        if (original_width > bound_width) {
+            //scale width to fit
+            new_width = bound_width;
+            //scale height to maintain aspect ratio
+            new_height = (new_width * original_height) / original_width;
+        }
+    
+        // then check if we need to scale even with the new height
+        if (new_height > bound_height) {
+            //scale height to fit instead
+            new_height = bound_height;
+            //scale width to maintain aspect ratio
+            // The '* 125 / 100' part is an embarrassing hack to make scaled images 25% wider
+            // since I couldn't figure out what was shrinking them by 25% with the amount of time I had
+            new_width = (new_height * original_width * 125 / 100) / original_height;
```

Comment:
hack

Reply:
Created issue https://github.com/kumoricon/registration/issues/161 to resolve the root cause of this